lib/enigma.rb

```diff
@@ -11,18 +12,25 @@ def self.characters
     @@characters
   end
 
-  def encrypt(message, key = rand(10000), date = Time.now)
-    @key = format('%05d', key)
-    @date = date.strftime('%m%d%y')
+  def add_offset
+    finder = (@date.to_i ** 2).to_s
+    @offset = format('%04d',finder[-4..-1])
+  end
 
+  def encrypt(message, key = format('%05d', rand(10000)), date = Time.now.strftime('%m%d%y'))
+    accumulator = {}
+    @key = key
+    @date = date
     add_offset
     change = Shift.new(@key, @offset)
     change.make_shift
     @shift = change.shift
-  end
 
-  def add_offset
-    finder = (@date.to_i ** 2).to_s
-    @offset = format('%04d',finder[-4..-1])
+    @message = Message.new(message, @shift)
+    accumulator[:encryption] = @message.change_message
+    accumulator[:key] = @key
+    accumulator[:date] = @date
+    accumulator
   end
+
 end
```

Comment:
Create an accumulator to create a final hash for encrypt method.  Integrates both message and shift classes to make shift hash and encrypt given strings.